compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/SymbolFactory.java

```diff
@@ -364,10 +364,9 @@ public BallerinaVariableSymbol createVariableSymbol(BVarSymbol symbol, String na
     }
 
     private boolean isReadonlyIntersectionArrayType(BType type) {
-        if (type.tag == TypeTags.INTERSECTION && type.tsymbol.getOrigin() == SymbolOrigin.VIRTUAL) {
+        if (type.tag == TypeTags.INTERSECTION && type.tsymbol != null && type.tsymbol.getOrigin() == SymbolOrigin.VIRTUAL) {
```

Comment:
This may exceed # of chars for the horizontal line. Shall we break this into two lines?

Reply:
Will do